### PR TITLE
Update agave dependencies, switch to more-maintained rust toolchain GHA.

### PR DESCRIPTION
**Pull request by nlgripto**
Richie fixed another panic in the vote program.

### CODE REVIEW

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -11,7 +11,7 @@ jobs:
     runs-on: ubuntu-latest
     steps:
       - uses: actions/checkout@v4
-      - uses: actions-rust-lang/setup-rust-toolchain@v1
+      - uses: dtolnay/rust-toolchain@1.76.0
```
Comment (nlgripto) [inline .github/workflows/build.yml:14]: maintainer has said multiple times he does not GAF about rust-toolchain.toml

File: src/lib.rs [hunk 2]
```diff
@@ -562,27 +561,35 @@ fn execute_instr(input: InstrContext) -> Option<InstrEffects> {
...
+            /* pub fn load_program_with_pubkey<CB: TransactionProcessingCallback, FG: ForkGraph>(
+    callbacks: &CB,
+    program_cache: &ProgramCache<FG>,
+    pubkey: &Pubkey,
+    slot: Slot,
+    effective_epoch: Epoch,
+    epoch_schedule: &EpochSchedule,
+    reload: bool,
+) -> Option<Arc<ProgramCacheEntry>> { */
             if let Some(loaded_program) =
```
Comment (topointon-jump) [inline src/lib.rs:584]: Great to see comments! :D

One approach thing we've found which works really well is putting GitHub links to the Agave source code, to explain where logic comes from. See example: https://github.com/firedancer-io/firedancer/blob/vm-belt-sanding-public/src/flamenco/vm/syscall/fd_vm_syscall_hash.c

**Review by topointon-jump — APPROVED** [review]

**Review by topointon-jump — APPROVED** [review]
Thanks for doing this!

**Review by topointon-jump — APPROVED** [review]

**Comment by nlgripto** [comment]
@ripatel-fd I switched it to the one you recommended

**Comment by nlgripto** [comment]
https://github.com/firedancer-io/solfuzz-agave/pull/15#discussion_r1594915422

**Pull request merged**